### PR TITLE
dev/translation#34 Fix Contact Reference option for Postal Code

### PR DESCRIPTION
**Pull request by mlutfy**
Overview
----------------------------------------

To reproduce:

* Go to: https://dmaster.demo.civicrm.org/civicrm/admin/setting/search?reset=1
  * Under "Contact Reference Options", enable "Postal Code"
* Then go to Custom Fields for Individuals and add a new Custom Field of type "contact reference"
* Now edit an individual, and notice how the contact-reference field displays the country instead of the postcode.

Gitlab: https://lab.civicrm.org/dev/translation/issues/34

Technical Details
----------------------------------------

Seems like a copy-paste error in  xml/templates/civicrm_data.tpl.

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
@mlutfy I can replicate the bug but it's not fixed on the demo site - for the PR

http://core-16352-5cbt0.test-3.civicrm.org:8001/civicrm/contact/view?reset=1&cid=203

Screen Shot 2020-01-23 at 2 46 26 PM


http://core-16352-5cbt0.test-3.civicrm.org:8001/civicrm/admin/setting/search?reset=1




I think the  issue is that the generated file is not updated - @seamuslee001 we run setup.sh yeah?

File: CRM/Upgrade/Incremental/php/FiveTwentyTwo.php [hunk 1]
```diff
@@ -52,21 +52,47 @@ public function setPostUpgradeMessage(&$postUpgradeMessage, $rev) {
...
-  //  }
+  /**
+   * Upgrade function.
+   *
+   * @param string $rev
+   */
+  public function upgrade_5_22_alpha1($rev) {
+    $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addTask('dev/translation#34 Fix contact-reference option for Postal Code', 'fixContactRefOptionPostalCode');
+  }
```
Comment (demeritcowboy) [inline CRM/Upgrade/Incremental/php/FiveTwentyTwo.php]: The entire upgrade script should be in FiveTwentyThree.php.

**Comment by demeritcowboy** [comment]
For setup.sh/regen.sh to make installs with sample data work I think for this one it would be ok to just edit civicrm_generated.mysql directly since you're not affecting any id numbers. The part in (huge) line 1018 that has `(194,46,'Postal Code','8','country',NULL,0,NULL,8,NULL,0,0,1,NULL,NULL,NULL,NULL,NULL)`.

File: CRM/Upgrade/Incremental/php/FiveTwentyTwo.php (hunk 1 above)
Comment (mlutfy) [reply CRM/Upgrade/Incremental/php/FiveTwentyTwo.php]: Fixed, thanks!

**Comment by mlutfy** [comment]
@demeritcowboy Thanks for the hint regarding civicrm_generated.sql. I updated my PR.

**Comment by demeritcowboy** [comment]
Thanks I can give it a spin. I guess for the upgrade task it's hard to robustly find the record. Given the nature of the issue I don't think it's worth changing anything, but I'm curious in general about a multilanguage site where the upgrade is run by command line. I'm not sure how that works with the ts() - I guess I'll find out.

I'm not suggesting it for this PR but as a general technique what I'd probably do is write the function so it can also be called dry-run from the pre-upgrade check, and if it can't find the record the pre-upgrade would display a message sending them to the lab ticket. (I've learned that you can't display or queue post-upgrade messages during task runs, you can only abort the whole upgrade at that point.)

**Comment by mlutfy** [comment]
> I'm curious in general about a multilanguage site where the upgrade is run by command line. I'm not sure how that works with the ts() - I guess I'll find out.

It should still work in multilingual, because the option value label should be translated. The upgrade will run in the default site language, and then it will match `ts('Postal Code')` against `label_fr_CA`, for example.

> "as a general technique what I'd probably do is write the function so it can also be called dry-run from the pre-upgrade check, and if it can't find the record the pre-upgrade would display a message sending them to the lab ticket."

For important things, I would definitely agree.

Tangent: Failed test: ' api.v4.Entity.ConformanceTest.testConformance'

**Comment by seamuslee001** [comment]
@mlutfy that test fail is a now flaky one

**Comment by demeritcowboy** [comment]
* General standards
    * [`r-explain`] __PASS__
    * [`r-user`] __PASS__
    * [`r-doc`] __PASS__
    * [`r-run`] __Issue but...__
      * given the nature of the issue and how simple it is to fix manually in the database, this should be good as-is, and even if it misses some upgrades it will work for all new installs with and without sample data. I can see a couple language-related scenarios where it skips on upgrade, for example if you install it in english, then keep it single-lingual but switch to french, then upgrade. Similarly for if you install in english, then go multi but change default lang to non-english.
* Developer standards
    * [`r-tech`] __PASS__
    * [`r-code`] __PASS__
    * [`r-maint`] __N/A__
    * [`r-test`] __PASS__

**Comment by eileenmcnaughton** [comment]
test fail is unrelated - agree that nothing above is blocking

**Pull request merged**